### PR TITLE
[TEST] Enable test when boost not found.

### PR DESCRIPTION
**Pull request by MediosZ**
When boost is not installed, it will be built from the source. But setting `set(BUILD_TESTING OFF)` will disable all subsequent tests.  This can be observed in Fedora CI and WIndows CI.
So we need a better way to disable the boost tests.

### CODE REVIEW

**Comment by juntao** [comment]
Hello, I am a [code review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

**Comment by MediosZ** [comment]
Actually, I don't get why adding `set(BUILD_TESTING OFF)` there. Since the unit tests inside boost are not executed by default during `FetchContent_MakeAvailable`.
Maybe it would be better just to delete [that line](https://github.com/WasmEdge/WasmEdge/blob/e0e6dcc6847595ddf3cb152836c7b89578d1da1c/CMakeLists.txt#L56).

File: CMakeLists.txt [hunk 1]
```diff
@@ -53,7 +53,7 @@ else()
   )
   set(BOOST_ENABLE_CMAKE ON)
   set(BOOST_RUNTIME_LINK static)
-  set(BUILD_TESTING OFF)
+  set(BUILD_TESTING ON)
```
Comment (hydai) [inline CMakeLists.txt:56]: How about disabling this here and enabling it again after the boost part?

**Review by hydai — CHANGES_REQUESTED** [review]

**Comment by hydai** [comment]
> Actually, I don't get why adding `set(BUILD_TESTING OFF)` there. Since the unit tests inside boost are not executed by default during `FetchContent_MakeAvailable`. Maybe it would be better just to delete [that line](https://github.com/WasmEdge/WasmEdge/blob/e0e6dcc6847595ddf3cb152836c7b89578d1da1c/CMakeLists.txt#L56).

Oh, so the Fetch and build from source will not run the test part inside boost. If so, then we can remove it.

**Review by hydai — APPROVED** [review]

**Pull request merged**